### DISCUSSION TITLE
Next.js SSR getInitialProps not working on Vercel Production Deployment?

### DISCUSSION THREAD

**Discussion by kumarabhirup** [vercel/next.js#24418] [Help] [answered]
Hey all,
was working on this 👉 [https://github.com/KumarAbhirup/stripe-connect-demo](https://github.com/KumarAbhirup/stripe-connect-demo).

When that code is run on the development server, server-side-rendering works perfectly fine,
but when [deployed on Vercel](https://stripe-connect-demo.vercel.app), `getInitialProps` doesn't seem to run.

When people connect their Stripe account, this is supposed to be seen.
![Screenshot 2021-04-24 at 3 42 39 PM](https://user-images.githubusercontent.com/25429790/115956671-c6074f80-a51b-11eb-9049-9b7cf9bc3a60.jpg)

It works as expected on the development server, 
but when deployed with correct `.env` variables, by running `vercel` command,
it deploys and all, but the `getInitialProps` doesn't work, 
and therefore the page can't fetch and looks all blank.

### Here's the main logic...
```ts
import React from 'react';
import fetch from 'isomorphic-unfetch';

import HomeView from '@/views/Home';

const Home: React.FC<{
  data?: Record<string, unknown>;
  req: Record<string, unknown>;
}> = (props) => {
  return <HomeView data={props} />;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
Home.getInitialProps = async ({ req }: { req: { __NEXT_INIT_QUERY: any } }) => {
  const body = req?.__NEXT_INIT_QUERY;

  let response;
  try {
    response = await fetch(
      process.env.NEXT_PUBLIC_BASE_URL + '/api/verifyStripe',
      {
        method: 'POST',
        body: JSON.stringify(body),
        headers: {
          'Content-Type': 'application/json',
        },
      },
    ).then((res) => res.json());
  } catch (error) {
    return { data: { error }, req: body };
  }

  return { data: response, req: body };
};

export default Home;
```

Thanks in advance!

**Comment by bobarros** [comment, marked as answer]
No problem here. 
Double-check that you changed this to the production URL:

```
NEXT_PUBLIC_BASE_URL=http://localhost:3001
```

And also added the same URL to Stripe as a possible URI redirect.

**Reply by kumarabhirup** [reply]
Hi! I doubled checked, I did both of those things! It still doesn't seem to hydrate the client on production. 
This time I tried `getServerSideProps`.

**Reply by bobarros** [reply]
I had no problem using your repo with Vercel in production. 
There's some stripe or env config that it's not correct.

**Reply by kumarabhirup** [reply]
I see, thanks for the help! I filled the .env from Vercel GUI dashboard and now it seems to work.
